source/tutorials/learning-journey/sharing-dependencies.md

```diff
@@ -0,0 +1,198 @@
+
+One solution could be to add those packages to your build.
+This would certainly work in a pinch, but now your build depends on packages that aren't necessary for it to actually build.
+A better solution is to add those development packages to a shell environment so that the build dependencies stay as lean as possible.
+
+However, now you need to define a `shell.nix` that not only provides your development packages, but can also build your project.
+In other words, you need a `shell.nix` that brings in all of the packages that your build depends on.
+You could certainly copy the build dependencies from `default.nix` and copy them into `shell.nix`, but this is less than ideal.
+Your build dependencies are defined in multiple places, and aside from repeating yourself there's now the possiblity that the dependencies in `default.nix` and `shell.nix` may fall out of sync.
+
+There is a better way!
+
+## Getting started
+
+Create a directory called `shared_project` and enter it:
+```console
+$ mkdir shared_project
+$ cd shared_project
+```
+
+You'll be creating a Python web application as an example project, but don't worry, you'll be given all of the code you need and won't need to know Python to proceed.
+
+Create a file `app.py` with the following contents:
+```python
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route("/")
+def hello_world():
+    return "<p>Hello, World!</p>"
+```
+This creates a web application that returns `<p>Hello, World!</p>` on the `/` route.
+
+Next create a `pyproject.toml` file with the following contents:
+```toml
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "shared_project"
+version = "0.0.1"
+```
+This file tells Python how to build the project.
+
+For the Nix part of the project you'll create two files: `build.nix` and `default.nix`.
+The actual build recipe will be in `build.nix` and `default.nix` will import this file to perform the build.
+
+First create a `build.nix` file like this:
+```nix
+{
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication {
+  pname = "shared_project";
+  version = "0.0.1";
+  format = "pyproject";
+  src = builtins.path { path = ./.; name = "shared_project_source"; };
+  propagatedBuildInputs = with python3Packages; [
+    setuptools-scm
+    flask
+  ];
+}
+```
+The Nix expression in this file is a function that takes `python3Packages` as input and produces a derivation.
+This method of defining builds is a common design pattern in the Nix community, and is the format used throughout the `nixpkgs` repository.
+This particular derivation builds your Python application and ensures that `flask`, the library used to create the web application, is available at runtime for the application.
+
+Finally, create a `default.nix` that looks like this:
+```nix
+let
+  pkgs = import <nixpkgs> {};
```

Comment:
Good practice:
```suggestion
  pkgs = import <nixpkgs> {
    # Avoids impurities
    config = {};
    overlays = [];
  };
```

Reply:
Also avoid angle brackets. This will also prevent accidental CI breakages and automatically piling up stale store paths once we start testing code samples.